### PR TITLE
T05: Barebone Tablet mode

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #1572

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 1]
```diff
@@ -474,29 +464,23 @@ class ThreadFragment : Fragment() {
         val shouldLoadDistantResources = shouldLoadDistantResources(message.uid)
 
         if (message.getRecipientsForReplyTo(replyAll = true).second.isEmpty()) {
-            safeNavigateToNewMessageActivity(
+            goToNewMessageActivity(
                 draftMode = DraftMode.REPLY,
                 previousMessageUid = message.uid,
                 shouldLoadDistantResources = shouldLoadDistantResources,
+                arrivedFromExistingDraft = false,
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:471]: Useless

File: app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt [hunk 2]
```diff
@@ -196,6 +199,24 @@ class MainViewModel @Inject constructor(
...
+    val currentThreadUid = MutableLiveData<String?>(null)
+
+    val isInThread get() = currentThreadUid.value != null
+    val closeThreadTrigger = SingleLiveEvent<Unit>()
+
+    val downloadAttachmentsArgs = SingleLiveEvent<Triple<String, String, AttachmentType>>()
+    val newMessageArgs = SingleLiveEvent<NewMessageActivityArgs>()
+    val replyBottomSheetArgs = SingleLiveEvent<Pair<String, Boolean>>()
+    val threadActionsBottomSheetArgs = SingleLiveEvent<Triple<String, String, Boolean>>()
+    val messageActionsBottomSheetArgs = SingleLiveEvent<MessageActionsArgs>()
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt:212]: We should create dataclass to encapsulate each args, put them in a separate file

File: app/src/main/java/com/infomaniak/mail/utils/Extensions.kt [hunk 3]
```diff
@@ -290,7 +291,8 @@ fun Fragment.navigateToThread(
             )
         }
     } else {
-        safeNavigate(R.id.threadFragment, ThreadFragmentArgs(thread.uid).toBundle())
+        val uid = thread?.uid ?: threadUid ?: return@runCatchingRealm
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/utils/Extensions.kt:294]: Put a com to explicit a bit this part

File: app/src/main/res/navigation/main_navigation.xml [hunk 4]
```diff
@@ -54,6 +51,14 @@
             android:id="@+id/action_threadListFragment_to_invalidPasswordFragment"
             app:destination="@id/invalidPasswordFragment" />
 
+        <!-- Region Notification: Click to open Thread -->
+        <argument
+            android:name="openThreadUid"
+            android:defaultValue="@null"
+            app:argType="string"
+            app:nullable="true" />
+        <!-- Endregion -->
```
Comment (FabianDevel) [inline app/src/main/res/navigation/main_navigation.xml:60]: Move the arguments before the actions

File: app/src/main/res/navigation/main_navigation.xml [hunk 5]
```diff
@@ -54,6 +51,14 @@
             android:id="@+id/action_threadListFragment_to_invalidPasswordFragment"
             app:destination="@id/invalidPasswordFragment" />
 
+        <!-- Region Notification: Click to open Thread -->
```
Comment (FabianDevel) [inline app/src/main/res/navigation/main_navigation.xml:54]: Uniformize region name with the others

File: gradle/libs.versions.toml [hunk 6]
```diff
@@ -38,6 +39,7 @@ lottie = { module = "com.airbnb.android:lottie", version.ref = "lottie" }
 navigation-safeargs = { module = "androidx.navigation:navigation-safe-args-gradle-plugin", version.ref = "navigation" }
 realm-kotlin-base = { module = "io.realm.kotlin:library-base", version.ref = "realmKotlin" }
 sentry-android-fragment = { module = "io.sentry:sentry-android-fragment", version.ref = "sentryAndroidFragment" }
+slidingpanelayout = { module = "androidx.slidingpanelayout:slidingpanelayout", version.ref = "slidingpanelayout" }
```
Comment (FabianDevel) [inline gradle/libs.versions.toml:42]: Rename this to snakeCase but keep the camel for the version

File: app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt (hunk 2 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt:212]: Will be done in another PR.

File: app/src/main/res/navigation/main_navigation.xml (hunk 4 above)
Comment (KevinBoulongne) [reply app/src/main/res/navigation/main_navigation.xml:60]: Will be done in another PR.

File: app/src/main/res/navigation/main_navigation.xml (hunk 5 above)
Comment (KevinBoulongne) [reply app/src/main/res/navigation/main_navigation.xml:54]: Will be done in another PR.

File: gradle/libs.versions.toml (hunk 6 above)
Comment (KevinBoulongne) [reply gradle/libs.versions.toml:42]: Ok for the camel, but not ok for the snake.

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListFragment.kt [hunk 7]
```diff
@@ -164,6 +163,23 @@ class ThreadListFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener {
...
+                        navigateToThread(mainViewModel, threadUid = openThreadUid)
+                        arguments?.remove(navigationArgs::openThreadUid.name)
+                        threadListObserver?.let(threadListAdapter::unregisterAdapterDataObserver)
+                    }
+                }
+            }
+        }
+        threadListObserver?.let(threadListAdapter::registerAdapterDataObserver)
+    }
+
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListFragment.kt:182]: Optimize the opening of notifications like the commit in `opti-opening-notifications` does so we don't wait on the adapter for no reason. Probably best to put it in its own PR though

File: app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt [hunk 8]
```diff
@@ -196,6 +199,24 @@ class MainViewModel @Inject constructor(
     }
     //endregion
 
+    //region Current Thread
+    val currentThreadUid = MutableLiveData<String?>(null)
+
+    val isInThread get() = currentThreadUid.value != null
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt:205]: `isInThread` is unused in this PR, is it normal?

**Review by LunarX — CHANGES_REQUESTED** [review]
Not finished reviewing but here some first comments for now

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadViewModel.kt [hunk 9]
```diff
@@ -74,31 +76,38 @@ class ThreadViewModel @Inject constructor(
...
-        messageController.getSortedAndNotDeletedMessagesAsync(threadUid)
-            ?.map { results -> results.list.map { splitBody(it) } }
-            ?.asLiveData()
-            ?.let { emitSource(it) }
+            splitBodies.clear()
+
+            messageController.getSortedAndNotDeletedMessagesAsync(threadUid)
+                ?.map { results -> results.list.map { splitBody(it) } }
+                ?.collect(messagesLive::postValue)
+        }
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadViewModel.kt:107]: This method is a bit complicated to understand. Maybe something like this? 

```kotlin
fun reassignMessagesLive(threadUid: String) {
        messagesLiveJob?.cancel()
        messagesLiveJob = viewModelScope.launch(ioCoroutineContext) {

            suspend fun splitBody(message: Message): Message = withContext(ioDispatcher) {
                val splitBodies = mutableMapOf<String, SplitBody>()

                if (message.body == null) return@withContext message

                message.apply {
                    body?.let {
                        val isNotAlreadySplit = !splitBodies.contains(message.uid)
                        if (isNotAlreadySplit) splitBodies[message.uid] = MessageBodyUtils.splitContentAndQuote(it)
                        splitBody = splitBodies[message.uid]
                    }
                }

                return@withContext message
            }

            messageController.getSortedAndNotDeletedMessagesAsync(threadUid)
                ?.map { results -> results.list.map { splitBody(it) } }
                ?.collect(messagesLive::postValue)
        }
    }
```

Also, you don't need that splitBodies variable to be a class variable. It could be within the inner function  ? Like in my suggestion ?

File: app/src/main/res/layout/fragment_search.xml [hunk 10]
```diff
@@ -15,181 +15,195 @@
...
     android:layout_width="match_parent"
-    android:layout_height="match_parent">
+    android:layout_height="match_parent"
+    tools:context=".ui.main.search.SearchFragment">
 
-    <com.google.android.material.appbar.AppBarLayout
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content">
+    <androidx.coordinatorlayout.widget.CoordinatorLayout
+        android:layout_width="600dp"
```
Comment (tevincent) [inline app/src/main/res/layout/fragment_search.xml:27]: Are we sure we want 600dp on all resolution and all size of tablet ?

File: app/src/main/res/layout/fragment_search.xml [hunk 11]
```diff
@@ -15,181 +15,195 @@
...
-            android:layout_height="wrap_content"
-            android:background="@color/backgroundColor"
-            android:clipToPadding="false"
-            android:paddingHorizontal="@dimen/marginStandardMedium"
-            android:scrollbars="none">
-
-            <LinearLayout
-                android:layout_width="wrap_content"
+            <HorizontalScrollView
+                android:id="@+id/horizontalScrollViewDummyId"
```
Comment (tevincent) [inline app/src/main/res/layout/fragment_search.xml:57]: Is that id necessary ?

File: app/src/main/res/layout/fragment_thread.xml [hunk 12]
```diff
@@ -22,107 +22,122 @@
     android:layout_height="match_parent"
     tools:context=".ui.main.thread.ThreadFragment">
 
-    <com.google.android.material.appbar.MaterialToolbar
-        android:id="@+id/toolbar"
+    <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (tevincent) [inline app/src/main/res/layout/fragment_thread.xml:25]: Here you have a `ConstraintLayout` with two `ConstraintLayout`s inside. `ConstraintLayout`s are meant to avoid views intrication so here you might be able to just get rid of the two inner `ConstraintLayout`s ?

File: app/src/main/res/layout/fragment_thread.xml [hunk 13]
```diff
@@ -22,107 +22,122 @@
...
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/threadCoordinatorLayout"
+            app:menu="@menu/message_menu" />
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
+    <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (tevincent) [inline app/src/main/res/layout/fragment_thread.xml:136]: A `ConstraintLayout` for an empty view ?

File: app/src/main/res/layout/fragment_thread_list.xml [hunk 14]
```diff
@@ -15,275 +15,289 @@
...
-                            android:text="@string/buttonInstall"
-                            app:icon="@drawable/ic_app_update"
-                            app:iconPadding="@dimen/marginStandardSmall"
-                            app:iconSize="@dimen/marginStandardMedium"
-                            app:layout_constraintEnd_toEndOf="parent"
-                            app:layout_constraintTop_toTopOf="parent" />
-
-                        <com.google.android.material.divider.MaterialDivider
-                            android:id="@+id/installUpdateDivider"
+                        <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (tevincent) [inline app/src/main/res/layout/fragment_thread_list.xml:201]: Maybe you can only have the `ConstraintLayout` here instead of a `ConstraintLayout` inside a `FrameLayout` ?

**Review by tevincent — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 15]
```diff
@@ -265,6 +248,18 @@ class ThreadFragment : Fragment() {
         phoneContextualMenuAlertDialog.initValues(mainViewModel.snackBarManager)
     }
 
+    private fun observeThreadOpening() = with(binding) {
+
+        mainViewModel.currentThreadUid.distinctUntilChanged().observeNotNull(viewLifecycleOwner) { threadUid ->
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:251]: We can use only one LiveData, but the logic will still be split because it won't be observed in the same Fragment.
Here, it's the ThreadFragment, but the closing of the Thread is in the ThePaneFragment.

File: app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt [hunk 16]
```diff
@@ -373,7 +373,6 @@ class MainActivity : BaseActivity() {
     private fun setupSnackBar() {
         fun getAnchor(): View? = when (navController.currentDestination?.id) {
             R.id.threadListFragment -> findViewById(R.id.newMessageFab)
-            R.id.threadFragment -> findViewById(R.id.quickActionBar)
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:376]: True.
And in the Search, the Snackbar was totally on the quickActionBar.

It's fixed here: https://github.com/Infomaniak/android-kMail/pull/1597

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 17]
```diff
@@ -265,6 +246,18 @@ class ThreadFragment : Fragment() {
...
+
+            // Display Thread view
+            emptyView.isGone = true
+            threadView.isVisible = true
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:255]: Done !

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 18]
```diff
@@ -367,25 +358,28 @@ class ThreadFragment : Fragment() {
...
+            updateUi(threadUid, folderRole = mainViewModel.getActionFolderRole(result.thread))
+            updateAdapter(result)
         }
     }
 
-    private fun updateUi(folderRole: FolderRole?) = with(binding) {
+    private fun updateUi(threadUid: String, folderRole: FolderRole?) = with(binding) {
 
         iconFavorite.setOnClickListener {
             trackThreadActionsEvent(ACTION_FAVORITE_NAME, isFavorite)
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt]: Done !

File: app/src/main/java/com/infomaniak/mail/utils/Extensions.kt [hunk 19]
```diff
@@ -273,11 +273,18 @@ fun Fragment.safeNavigateToNewMessageActivity(args: Bundle? = null, currentClass
...
 }
 
-fun Fragment.navigateToThread(
+/**
+ * The `thread` is the default parameter for this function. But if we are coming from a Notification,
+ * we only have the `threadUid`. Since the `thread` is only used to know if it's `isOnlyOneDraft` or
+ * not, and that it can't be `true` if coming from a Notification, we can bypass the usage of the
+ * `thread` and directly use `threadUid` instead.
+ */
+fun TwoPaneFragment.navigateToThread(
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/utils/Extensions.kt:282]: All `navigate` functions are purposely in the same place to be easier to find.

File: app/src/main/java/com/infomaniak/mail/utils/Extensions.kt [hunk 20]
```diff
@@ -273,11 +273,18 @@ fun Fragment.safeNavigateToNewMessageActivity(args: Bundle? = null, currentClass
...
     mainViewModel: MainViewModel,
-    thread: Thread,
+    thread: Thread? = null,
+    threadUid: String? = null,
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/utils/Extensions.kt:285]: Done here: https://github.com/Infomaniak/android-kMail/pull/1595?w=1

File: app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt (hunk 8 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt]: Oversight, it's fixed.

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt [hunk 21]
```diff
@@ -0,0 +1,137 @@
...
+import com.infomaniak.lib.core.utils.safeNavigate
+import com.infomaniak.mail.R
+import com.infomaniak.mail.ui.MainViewModel
+import com.infomaniak.mail.ui.main.search.SearchFragment
+import com.infomaniak.mail.ui.main.thread.DetailedContactBottomSheetDialogArgs
+import com.infomaniak.mail.ui.main.thread.ThreadFragment
+import com.infomaniak.mail.ui.main.thread.actions.*
+import com.infomaniak.mail.utils.safeNavigateToNewMessageActivity
+
+open class TwoPaneFragment : Fragment() {
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt:35]: "two-pane" here is an compound adjective created with a number and a noun, so it should use the singular form of the noun.

And it's someone named _"Bodin"_ that said it, so it's definitively true :
https://yolainebodin.com/the-language-nook/english/compound-adjective-with-number-and-noun

=> _"Careful! Remember to keep the noun in the singular form even if the number makes you feel there should be an s for the plural because, as you know, it is now an adjective and there is no s at the end of adjectives in English!"_

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt [hunk 22]
```diff
@@ -0,0 +1,137 @@
...
+
+    private val slidingPaneLayout: SlidingPaneLayout
+        get() = when (this) {
+            is ThreadListFragment -> binding.threadListSlidingPaneLayout
+            is SearchFragment -> binding.searchSlidingPaneLayout
+            else -> throw IllegalStateException()
+        }
+
+    fun areBothShown() = !slidingPaneLayout.isSlideable
+    fun isOnlyLeftShown() = slidingPaneLayout.let { it.isSlideable && !it.isOpen }
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt:47]: These are helpers that will be used in a following PR.

File: app/src/main/res/layout/fragment_search.xml (hunk 11 above)
Comment (KevinBoulongne) [reply app/src/main/res/layout/fragment_search.xml:57]: No clue, you should see with @LunarX , he added it.

File: app/src/main/res/layout/fragment_thread.xml (hunk 13 above)
Comment (KevinBoulongne) [reply app/src/main/res/layout/fragment_thread.xml:136]: We know we want to put something in here, but we don't have any design yet.

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadViewModel.kt (hunk 9 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadViewModel.kt:107]: True, `splitBodies` doesn't need to be a class variable, but we can't put it in `fun splitBody()`.
If we do that, it will reset each time `splitBody()` is called, defeating its purpose.

Instead, we can put it in `reassignMessagesLive ()`, just after launching the coroutine.

File: app/src/main/res/layout/fragment_thread_list.xml (hunk 14 above)
Comment (KevinBoulongne) [reply app/src/main/res/layout/fragment_thread_list.xml:201]: This is simplified in a following PR : https://github.com/Infomaniak/android-kMail/pull/1579

Comment (KevinBoulongne) [reply app/src/main/res/layout/fragment_thread_list.xml:201]: Noted, will be handled in a foreseeable future, since it's not tied to the Tablet mode feature.

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt (hunk 21 above)
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt:35]: This class is not meant to be instanciated, it should be abstract

**Review by FabianDevel — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/utils/Extensions.kt (hunk 19 above)
[the remark this answers is not in the document]
Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/utils/Extensions.kt:282]: Actually place it directly inside TwoPaneFragment, it's only gonna be used by this class

File: app/src/main/res/layout/fragment_search.xml (hunk 11 above)
Comment (LunarX) [reply app/src/main/res/layout/fragment_search.xml:57]: Yeah, by giving it an id, the scroll view will retain how much it has scrolled when the activity is recreated. Without the idea it's reset to 0 every time. We could add a comment nontheless

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt (hunk 15 above)
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:249]: Done !

File: app/src/main/java/com/infomaniak/mail/utils/Extensions.kt (hunk 19 above)
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/utils/Extensions.kt:282]: Done !

File: app/src/main/res/layout/fragment_thread_list.xml [hunk 23]
```diff
@@ -15,275 +15,289 @@
...
 
-    <com.google.android.material.appbar.AppBarLayout
-        android:id="@+id/appBarLayout"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toTopOf="parent">
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="600dp"
```
Comment (LunarX) [inline app/src/main/res/layout/fragment_thread_list.xml:27]: Do we really want to only start showing the two panes above 600 + 550 dp and not before? Also isn't 600dp a bit big for some smaller tablette widths?

Comment (LunarX) [reply app/src/main/res/layout/fragment_thread_list.xml:27]: Also this should be in a dimension if it is shared with another layout

**Review by LunarX — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/fragment_search.xml (hunk 11 above)
Comment (KevinBoulongne) [reply app/src/main/res/layout/fragment_search.xml:62]: Done.

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 24]
```diff
@@ -437,13 +422,7 @@ class ThreadFragment : Fragment() {
...
             startActivity(openWithIntent(requireContext()))
         } else {
-            safeNavigate(
-                ThreadFragmentDirections.actionThreadFragmentToDownloadAttachmentProgressDialog(
-                    attachmentResource = resource!!,
-                    attachmentName = name,
-                    attachmentType = getFileTypeFromMimeType(),
-                )
-            )
+            mainViewModel.downloadAttachmentsArgs.value = Triple(resource!!, name, getFileTypeFromMimeType())
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:425]: Does it make sens to turn this into a `SingleLiveEvent` rather than defining a method to call from here that directly executes what needs to be executed? Because it's only triggered from here and is only called once because it's a `SingleLiveEvent`, no?

Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:425]: We need to think about it but maybe it could even make sens to extract it to a separate use case file depending on the situation

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 25]
```diff
@@ -182,25 +170,21 @@ class ThreadFragment : Fragment() {
...
-        updateUi(folderRole)
     }
 
-    private fun initAdapter() = with(binding.messagesList) {
+    private fun setupAdapter() = with(binding.messagesList) {
         adapter = ThreadAdapter(
             shouldLoadDistantResources = shouldLoadDistantResources(),
             onContactClicked = { contact ->
-                safeNavigate(ThreadFragmentDirections.actionThreadFragmentToDetailedContactBottomSheetDialog(contact))
+                mainViewModel.detailedContactArgs.value = contact
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:179]: Same comment than the previous `downloadAttachmentsArgs` one. And actually the comment works probably for all of them

File: app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt [hunk 26]
```diff
@@ -196,6 +199,23 @@ class MainViewModel @Inject constructor(
...
+    val currentThreadUid = MutableLiveData<String?>(null)
+
+    val closeThreadTrigger = SingleLiveEvent<Unit>()
+
+    val downloadAttachmentsArgs = SingleLiveEvent<Triple<String, String, AttachmentType>>()
+    val newMessageArgs = SingleLiveEvent<NewMessageActivityArgs>()
+    val replyBottomSheetArgs = SingleLiveEvent<Pair<String, Boolean>>()
+    val threadActionsBottomSheetArgs = SingleLiveEvent<Triple<String, String, Boolean>>()
+    val messageActionsBottomSheetArgs = SingleLiveEvent<MessageActionsArgs>()
+    val detailedContactArgs = SingleLiveEvent<Recipient>()
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt:212]: Why were all of these extracted actually?

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 27]
```diff
@@ -265,6 +244,26 @@ class ThreadFragment : Fragment() {
...
     }
 
+    private fun observeThreadOpening() = with(threadViewModel) {
+
+        mainViewModel.currentThreadUid.distinctUntilChanged().observeNotNull(viewLifecycleOwner) { threadUid ->
+
+            reassignThreadLive(threadUid)
+            reassignMessagesLive(threadUid)
+
+            openThread(threadUid).observe(viewLifecycleOwner) { result ->
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:254]: This `openThread()` method needs the updated Thread object to work. Shouldn't we wait for `reassignThreadLive` to have finished before calling `openThread()` so we can skip on sending some of the `threadUid` values that are passed around everywhere and use the `threadLive` latest value instead?

File: app/src/main/res/layout/fragment_thread_list.xml
Comment (LunarX) [inline app/src/main/res/layout/fragment_thread_list.xml:1]: I will write this here so we can have a thread on github.

We should avoid having the sliding animation of the pane because this won't be compatible with the feature of swiping between emails when we implement it. If the email swipes into view from the right and when you swipe it back away from the screen it goes to another email instead of discarding the pane, it will be very confusing and counter intuitive. Also, because this would be a change of habit for the users, I don't think we want to force them to get used to this new sliding behavior for them to have to get used back to the "non sliding" behavior again when we can swipe between emails.

Currently the color of the status bar is very chaotic. I think this needs to be thought through. UI/UX should be involved if needed but this needs to be tackled for the tablet update to roll out imo. This has been postponed for long enough

**Review by LunarX — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/fragment_search.xml (hunk 10 above)
Comment (KevinBoulongne) [reply app/src/main/res/layout/fragment_search.xml:27]: Same question as https://github.com/Infomaniak/android-kMail/pull/1573?w=1#pullrequestreview-1779351753 , so I'll answer there.

File: app/src/main/res/layout/fragment_thread_list.xml (hunk 23 above)
Comment (KevinBoulongne) [reply app/src/main/res/layout/fragment_thread_list.xml:27]: Why 600dp ? Because of this : https://developer.android.com/guide/topics/large-screens/support-different-screen-sizes

The 500dp is a little more arbitrary. If we were totally following the previously linked documentation, we should put 240dp instead of 500dp, but on some tablets it's not enough (for exemple, on Pixel C).

With some trial and error, I found 500dp to be nice. The value itself is not really important, it just needs to be big enough to not show up when the tablet is in portrait, but while still being inferior to the tablet width.

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 28]
```diff
@@ -367,25 +362,15 @@ class ThreadFragment : Fragment() {
...
-
-    private fun updateUi(folderRole: FolderRole?) = with(binding) {
+        // Display Thread view
+        emptyView.isGone = true
+        threadView.isVisible = true
 
         iconFavorite.setOnClickListener {
-            trackThreadActionsEvent(ACTION_FAVORITE_NAME, isFavorite)
-            mainViewModel.toggleThreadFavoriteStatus(navigationArgs.threadUid)
+            trackThreadActionsEvent(ACTION_FAVORITE_NAME, threadViewModel.threadLive.value?.isFavorite ?: false)
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:372]: Indeed, it doesn't.

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt (hunk 24 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:425]: We need to navigate, and since we are in the ThreadFragment, we are out of navigation. So we can't do it here.

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt (hunk 25 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:179]: Answered here : https://github.com/Infomaniak/android-kMail/pull/1573?w=1#discussion_r1425256192

File: app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt (hunk 26 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt:212]: Answered here : https://github.com/Infomaniak/android-kMail/pull/1573#discussion_r1425256192

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt (hunk 24 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:425]: We can extract the `//region Current Thread` to a TwoPaneViewModel, to help lighten the MainViewModel.

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt (hunk 27 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:254]: How do you want to do that exactly? I don't see any elegant way.

File: app/src/main/res/layout/fragment_thread_list.xml
Comment (LunarX) [reply app/src/main/res/layout/fragment_thread_list.xml:1]: No that I see it, the statusbar issue also applies to the bottom navbar

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListFragment.kt (hunk 7 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListFragment.kt:182]: Fixed in https://github.com/Infomaniak/android-kMail/pull/1606?w=1

**Review by tevincent — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt (hunk 28 above)
[the remark this answers is not in the document]
Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt]: Now, does it really make sens to send a "favorite" matomo without any favorite value if it's null?

File: app/src/main/res/layout/fragment_thread_list.xml (hunk 23 above)
Comment (KevinBoulongne) [reply app/src/main/res/layout/fragment_thread_list.xml]: We still need to block TwoPane layout in portrait mode.
Right now, without handling configuration changes, it's impossible.
For now we block it until screens of 1100 width.

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 29]
```diff
@@ -158,10 +145,11 @@ class ThreadFragment : Fragment() {
         _binding = null
     }
 
-    private fun setupUi(folderRole: FolderRole?) = with(binding) {
+    private fun setupUi() = with(binding) {
 
         toolbar.setNavigationOnClickListener {
-            closeThread()
+            val onlyThreadIsShown = (parentFragment as TwoPaneFragment).isOnlyRightShown()
+            if (onlyThreadIsShown) mainViewModel.closeThread()
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:152]: Done in https://github.com/Infomaniak/android-kMail/pull/1592

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt (hunk 24 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:425]: Done here https://github.com/Infomaniak/android-kMail/pull/1606?w=1

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt (hunk 29 above)
[the remark this answers is not in the document]
Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:152]: This is actually modified inside https://github.com/Infomaniak/android-kMail/pull/1580 instead
